controller/program/tests/tests/rebalance/test_utils.rs

```diff
@@ -0,0 +1,502 @@
+            executable: false,
+            rent_epoch: 0,
+        },
+    )
+}
+
+pub fn mock_empty_rebalance_record_account() -> Account {
+    const RR_RENT: u64 = Rent::DEFAULT.min_balance(std::mem::size_of::<RebalanceRecord>());
+    Account {
+        lamports: RR_RENT,
+        data: vec![0; std::mem::size_of::<RebalanceRecord>()],
+        owner: Pubkey::new_from_array(SYSTEM_PROGRAM_ID),
+        executable: false,
+        rent_epoch: 0,
+    }
+}
+
+pub fn assert_start_success(
+    bef: &[PkAccountTup],
+    aft: &[PkAccountTup],
+    out_mint: &[u8; 32],
+    inp_mint: &[u8; 32],
+) {
+    let [pool_accounts, lst_state_accounts] = [POOL_STATE_ID, LST_STATE_LIST_ID]
+        .map(|pk| acc_bef_aft(&Pubkey::new_from_array(pk), bef, aft));
+
+    let [pool_bef, pool_aft] = pool_accounts.each_ref().map(|acc| {
+        PoolStatePacked::of_acc_data(&acc.data)
+            .expect("pool packed (before/after)")
+            .into_pool_state()
+    });
+
+    assert_diffs_pool_state(
+        &DiffsPoolStateArgs {
+            bools: NewPoolStateBoolsBuilder::start()
+                .with_is_rebalancing(Diff::StrictChanged(false, true))
+                .with_is_disabled(Diff::Pass)
+                .build(),
+            total_sol_value: Diff::Pass,
+            ..Default::default()
+        },
+        &pool_bef,
+        &pool_aft,
+    );
+
+    let [lst_state_list_bef, lst_state_list_aft]: [Vec<_>; 2] =
+        lst_state_accounts.each_ref().map(|acc| {
+            LstStatePackedList::of_acc_data(&acc.data)
+                .expect("lst state list packed")
+                .0
+                .iter()
+                .map(|packed| packed.into_lst_state())
+                .collect()
+        });
+
+    let diffs = LstStateListChanges::new(&lst_state_list_bef)
+        .with_diff_by_mint(
+            out_mint,
+            DiffLstStateArgs {
+                sol_value: Diff::Pass,
+                ..Default::default()
+            },
+        )
+        .with_diff_by_mint(
+            inp_mint,
+            DiffLstStateArgs {
+                sol_value: Diff::Pass,
+                ..Default::default()
+            },
+        )
+        .build();
+    assert_diffs_lst_state_list(&diffs, &lst_state_list_bef, &lst_state_list_aft);
+
+    let inp_lst_idx = lst_state_list_bef
+        .iter()
+        .position(|state| state.mint == *inp_mint)
+        .expect("input LST exists");
+
+    let rr_pk = Pubkey::new_from_array(REBALANCE_RECORD_ID);
+    let rr_acc = aft
+        .iter()
+        .find_map(|(pk, acc)| (*pk == rr_pk).then_some(acc))
+        .expect("rebalance record exists");
+
+    let rr =
+        unsafe { RebalanceRecord::of_acc_data(&rr_acc.data).expect("rebalance record unpack") };
+
+    assert_eq!(
+        rr.inp_lst_index, inp_lst_idx as u32,
+        "dst lst index mismatch"
+    );
+    assert!(
+        rr.old_total_sol_value > 0,
+        "old_total_sol_value should be captured post-sync"
+    );
```

Comment:
this should check that this is `== pool_bef.total_sol_value` + changes after a manual sync sol value of out lst because that might be out of sync